core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -1178,32 +1160,20 @@ private boolean stateNotActive() {
         return  partitionState() != SharePartitionState.ACTIVE;
     }
 
-    private void completeInitializationWithException() {
-        lock.writeLock().lock();
-        try {
-            partitionState = SharePartitionState.FAILED;
-        } finally {
-            lock.writeLock().unlock();
-        }
-    }
-
-    private InitializationResult checkInitializationCompletion() {
+    private boolean initializedOrThrowException() {
         SharePartitionState currentState = partitionState();
-        switch (currentState) {
-            case ACTIVE:
-                return new InitializationResult(true);
-            case FAILED:
-                return new InitializationResult(true, new IllegalStateException(String.format("Share partition failed to load %s-%s", groupId, topicIdPartition)));
-            case INITIALIZING:
-                return new InitializationResult(true, new LeaderNotAvailableException(String.format("Share partition is already initializing %s-%s", groupId, topicIdPartition)));
-            case FENCED:
-                return new InitializationResult(true, new FencedStateEpochException(String.format("Share partition is fenced %s-%s", groupId, topicIdPartition)));
-            case EMPTY:
-                // Do not complete the future as the share partition is not yet initialized.
-                return new InitializationResult(false);
-            default:
```

Comment:
Any particular reason why we've removed the default case now?

Reply:
Now code uses `switch type pattern` as minNonClientJavaVersion is set to 17, hence that doesn't require default on enum. If new enum type will be added in future but not is switch statement then compile time exception will occur in build.

Reply:
makes sense, then we can probably remove `ClassDataAbstractionCoupling` from `suppressions.xml` [here](https://github.com/apache/kafka/blob/trunk/checkstyle/suppressions.xml#L50C44-L50C72)

Reply:
Yeah, removed the ClassDataAbstractionCoupling suppression.